### PR TITLE
chore(linux): Update method comments

### PR DESCRIPTION
**Pull request by ermshiperete**
@keymanapp-test-bot skip

### CODE REVIEW

File: linux/ibus-keyman/src/keymanutil.c [hunk 1]
```diff
@@ -369,6 +368,7 @@ keyman_get_options_queue_fromdconf(gchar *package_id,
             }
             index++;
         }
+        g_strfreev(options);
```
Comment (mcdurdin) [inline linux/ibus-keyman/src/keymanutil.c:371]: I note on line 352 you have: 

```
    if (*options != NULL)
```

But `keyman_get_options_fromdconf` can return `NULL` so we should be testing for that, not `*options` surely.

**Review by mcdurdin — CHANGES_REQUESTED** [review]

File: linux/ibus-keyman/src/keymanutil.c (hunk 1 above)
Comment (ermshiperete) [reply linux/ibus-keyman/src/keymanutil.c:371]: Thanks for noticing. Done.

**Review by mcdurdin — COMMENTED** [review]
Yep, that looks better. Saw you fixed another location.

Okay, so now I am reading `keyman_put_options_todconf` and it looks like the reallocs in that are conflating an array-of-pointers with the array-containing-strings (which does not really exist in C++). See for example lines 422, 430. It's probably going to "work" much of the time, assuming `kvp` is longer than the existing options array size... but it's certainly not right and would cause grief sometimes...

**Comment by ermshiperete** [comment]
:+1: You're right, this doesn't work - it looked suspicious but since it's always been that way... :smile: I added unit tests and that showed that it doesn't work if it has to reallocate memory. Should be fixed now :smile:.

**Review by mcdurdin — APPROVED** [review]
LGTM

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.148-alpha](https://keyman.com/downloads/releases/alpha/15.0.148)

**Pull request merged**